### ISSUE TITLE
bindAndHandleAsync called with no `parallelism` argument doesn't work with Http2

### ISSUE THREAD

**Issue by jrudolph** [akka/akka-http#2168]
Also `maxConcurrentStreams` isn't adapted when a value parallelism setting is given.

Broken in 10.1.4.

**Comment by jrudolph** [comment]
`Http2BindingViaConfigSpec` should have caught this but didn't, I guess because it didn't run any requests (because of a missing client?).

**Comment by chbatey** [comment]
This results in the default value of 0 filtering through and resulting it being passed to a mapAsync:

```
[akka://akka-grpc-server-java/system/StreamSupervisor-240/flow-1-0-detacher] Error during preStart in [MapAsyncUnordered(0,akka.http.impl.engine.http2.Http2Blueprint$$$Lambda$750/955897381@3eb1bd1a)]: size must be positive
```

**Comment by raboof** [comment]
This is likely also a symptom of this issue, right?

```
java.lang.IllegalArgumentException: size must be positive
akka.stream.impl.FixedSizeBuffer$.apply(Buffers.scala:64)
akka.stream.impl.Buffer$.apply(Buffers.scala:45)
akka.stream.impl.Buffer$.apply(Buffers.scala:40)
akka.stream.impl.fusing.MapAsyncUnordered$$anon$26.preStart(Ops.scala:1277)
akka.stream.impl.fusing.GraphInterpreter.init(GraphInterpreter.scala:295)
```

**Comment by jrudolph** [comment]
Yep.

**Comment by jrudolph** [comment]
Fixed in #2146 a long time ago.
